src/vivarium_gates_mncnh/components/mortality.py

```diff
@@ -0,0 +1,167 @@
+
+    @property
+    def configuration_defaults(self) -> dict[str, Any]:
+        return {
+            self.name: {
+                "data_sources": {
+                    **{
+                        "life_expectancy": "population.theoretical_minimum_risk_life_expectancy"
+                    },
+                    **{
+                        f"{cause}_case_fatality_rate": partial(
+                            self.load_cfr_data, key_name=cause
+                        )
+                        for cause in self.maternal_disorders
+                    },
+                },
+            },
+        }
+
+    @property
+    def columns_created(self) -> list[str]:
+        return [COLUMNS.CAUSE_OF_DEATH, COLUMNS.YEARS_OF_LIFE_LOST]
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [
+            COLUMNS.ALIVE,
+            COLUMNS.EXIT_TIME,
+            COLUMNS.AGE,
+            COLUMNS.SEX,
+        ] + self.maternal_disorders
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def __init__(self) -> None:
+        super().__init__()
+        # TODO: update list of maternal disorders when implemented
+        self.maternal_disorders = [
+            COLUMNS.OBSTRUCTED_LABOR,
+            COLUMNS.MATERNAL_HEMORRHAGE,
+            COLUMNS.MATERNAL_SEPSIS,
+        ]
+
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.location = get_location(builder)
+
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        pop_update = pd.DataFrame(
+            {
+                COLUMNS.CAUSE_OF_DEATH: "not_dead",
+                COLUMNS.YEARS_OF_LIFE_LOST: 0.0,
+            },
+            index=pop_data.index,
+        )
+        self.population_view.update(pop_update)
+
+    def on_time_step(self, event) -> None:
+        if self._sim_step_name() != SIMULATION_EVENT_NAMES.MORTALITY:
+            return
+
+        pop = self.population_view.get(event.index)
+        has_maternal_disorders = pop[self.maternal_disorders]
+        has_maternal_disorders = has_maternal_disorders.loc[
+            has_maternal_disorders.any(axis=1)
+        ]
+
+        # Get raw and conditional case fatality rates for each simulant
+        choice_data = has_maternal_disorders.copy()
+        choice_data = self.calculate_case_fatality_rates(choice_data)
+
+        # Decide what simulants die from what maternal disorders
+        dead_idx = self.randomness.filter_for_probability(
+            choice_data.index,
+            choice_data["mortality_probability"],
+            "mortality_choice",
+        )
+
+        # Update metadata for simulants that died
+        if not dead_idx.empty:
+            pop.loc[dead_idx, COLUMNS.ALIVE] = "dead"
+
```

Comment:
@rmudambi Im leaving everyone as tracked but not sure if I should be doing that.

Reply:
Yes, that's correct. In general, we don't untrack simulants on death, but we especially want to keep them as tracked in this case, since the mother might die, but the child live, and we need to track that child in the neonatal period.